### PR TITLE
feat: apply sample rate

### PR DESCRIPTION
**Pull request by marandaneto**
## :loudspeaker: Type of change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
feat: apply sample rate

## :bulb: Motivation and Context
one should be able to sample rate events

## :pencil: Checklist

- [X] I reviewed submitted code
- [X] I added tests to verify changes
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

**Review by rxlabz — APPROVED** [review]
LGTM

File: dart/lib/src/client.dart [hunk 1]
```diff
@@ -45,6 +51,11 @@ abstract class SentryClient {
   }) async {
     event = _processEvent(event, eventProcessors: options.eventProcessors);
 
+    // dropped by sampling or event processors
+    if (event == null) {
+      return Future.value(SentryId.empty());
```
Comment (marandaneto) [inline dart/lib/src/client.dart:56]: @rxlabz should we use `Future.value(SentryId.empty());` or just `SentryId.empty()`?
cus the NoOp Client and Hub return `Future.value(...)` but the real implementation seems to return the String.

Comment (bruno-garcia) [reply dart/lib/src/client.dart:56]: It returns a Future so it wouldn't compile without this.

Or mark the method `async` but without an `await` in the body that wouldn't make sense

File: dart/lib/src/client.dart [hunk 2]
```diff
@@ -157,4 +176,11 @@ abstract class SentryClient {
     }
     return event;
   }
+
+  bool _sampleRate() {
```
Comment (bruno-garcia) [inline dart/lib/src/client.dart:180]: I think the should call `sample`, instead of `sampleRate`.

File: dart/lib/src/client.dart (hunk 1 above)
Comment (marandaneto) [reply dart/lib/src/client.dart:56]: It does compile, that's why the question, i guess the "compiler" infers that automatically.

**Pull request merged**